src/detect-icmpv6-mtu.c

```diff
@@ -0,0 +1,187 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Philippe Antoine <p.antoine@catenacyber.fr>
+ *
+ */
+
+#include "suricata-common.h"
+
+#include "detect.h"
+#include "detect-parse.h"
+
+#include "detect-icmpv6-mtu.h"
+#include "detect-util-uint.h"
+
+/* prototypes */
+static int DetectICMPv6mtuMatch (DetectEngineThreadCtx *, Packet *,
+        const Signature *, const SigMatchCtx *);
+static int DetectICMPv6mtuSetup (DetectEngineCtx *, Signature *, const char *);
+void DetectICMPv6mtuFree (void *);
+#ifdef UNITTESTS
+void DetectICMPv6mtuRegisterTests (void);
+#endif
+static int PrefilterSetupIcmpv6mtu(DetectEngineCtx *de_ctx, SigGroupHead *sgh);
+static bool PrefilterIcmpv6mtuIsPrefilterable(const Signature *s);
+
+/**
+ * \brief Registration function for icmpv6.mtu: keyword
+ */
+
+void DetectICMPv6mtuRegister(void)
+{
+    sigmatch_table[DETECT_ICMPV6MTU].name = "icmpv6.mtu";
+    sigmatch_table[DETECT_ICMPV6MTU].desc = "match on ICMPv6 MTU field";
+    sigmatch_table[DETECT_ICMPV6MTU].url = DOC_URL DOC_VERSION "/rules/header-keywords.html#icmpv6mtu";
+    sigmatch_table[DETECT_ICMPV6MTU].Match = DetectICMPv6mtuMatch;
+    sigmatch_table[DETECT_ICMPV6MTU].Setup = DetectICMPv6mtuSetup;
+    sigmatch_table[DETECT_ICMPV6MTU].Free = DetectICMPv6mtuFree;
+#ifdef UNITTESTS
+    sigmatch_table[DETECT_ICMPV6MTU].RegisterTests = DetectICMPv6mtuRegisterTests;
+#endif
+    sigmatch_table[DETECT_ICMPV6MTU].SupportsPrefilter = PrefilterIcmpv6mtuIsPrefilterable;
+    sigmatch_table[DETECT_ICMPV6MTU].SetupPrefilter = PrefilterSetupIcmpv6mtu;
+
+    DetectU32Register();
+    return;
+}
+
+// returns 0 on no mtu, and 1 if mtu
+static inline int packetGetIcmpv6MTU(Packet *p, uint32_t *picmpv6mtu) {
+    if (!(PKT_IS_ICMPV6(p)) || PKT_IS_PSEUDOPKT(p))
+        return 0;
+    if (!(ICMPV6_HAS_MTU(p)))
+        return 0;
+
+    *picmpv6mtu = ICMPV6_GET_MTU(p);
+    return 1;
+}
+
+/**
+ * \brief This function is used to match ICMPV6 MTU rule option on a packet with those passed via icmpv6.mtu:
+ *
+ * \param t pointer to thread vars
```

Comment:
`t` and `m` aren't parameters; `s` and `ctx` are missing.

Reply:
ok